llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -24150,6 +24150,13 @@ SDValue X86TargetLowering::LowerSELECT(SDValue Op, SelectionDAG &DAG) const {
                                           DAG.getBitcast(NVT, Op2)));
   }
 
+  // (select cc, 1.0, 0.0) -> (sint_to_fp (zext cc))
+  const ConstantFPSDNode *FPTV = dyn_cast<ConstantFPSDNode>(Op1);
+  const ConstantFPSDNode *FPFV = dyn_cast<ConstantFPSDNode>(Op2);
+  if (FPTV && FPFV && FPTV->isExactlyValue(1.0) && FPFV->isExactlyValue(0.0)) {
```

Comment:
This probably should be an optimization combine, and not in the lowering. Could also be in generic combiner if the constants aren't legal

Reply:
This could either be added to combineSelectOfTwoConstants (which is currently integer-only), or we start a similar combineSelectOfTwoFPConstants